util/rbac/rbac.go

```diff
@@ -188,10 +190,39 @@ func (e *Enforcer) EnableEnforce(s bool) {
 
 // LoadPolicy executes casbin.Enforcer functionality and will invalidate cache if required.
 func (e *Enforcer) LoadPolicy() error {
-	_, err := e.tryGetCabinEnforcer("", "")
+	_, err := e.tryGetCasbinEnforcer("", "")
 	return err
 }
 
+// CheckUserDefinedRoleReferentialIntegrity iterates over roles and policies to validate the existence of a matching policy subject for every defined role
+func CheckUserDefinedRoleReferentialIntegrity(e CasbinEnforcer) error {
+	allRoles, err := e.GetAllRoles()
```

Comment:
could we use [GetRolesForUser](https://github.com/casbin/casbin/blob/a98973beed58c2236ee453a126970f00d68c0f34/enforcer_interface.go#L64) and warn if the result is empty?

Reply:
The use case for this patch is to log a warning when a role mapping is defined without any policies which would reference it.  I was unable to find a way to get the casbin API to return all matching policies for a role.

I did some additional research based on this comment and realized that this check should also respect transitive mappings.  That is added in latest commit.

Reply:
Thanks for the tip- I refactored to use `GetImplicitPermissionsForUser` from the casbin API and checked for empty permissions.